File: umove-rider-ui/src/model/ride.ts
import { Vehicle } from './vehicle';
import { Zone } from './zone';
import { PaymentMethod } from './paymentmethod';
import { Rider } from './rider';
import { PaymentDetail } from './paymentdetail';
import { PromoCode } from './promocode';
import { Payment } from './payment';

export class Ride {
    _id: string;
    paymentDetail: PaymentDetail;
    payment: Payment;
    bookedAt: Date;
    status: string;
    vehicle: Vehicle;
    sourceZone: Zone;
    destinationZones: Zone[];
    paymentMethod: PaymentMethod;
    promocode: PromoCode;
    rider: Rider;
    distance: number;
    duration: number;
    rideStartAt: Date;
    rideEndAt: Date;
}

File: booking-service/src/main/java/in/stackroute/umove/bookingservice/model/Ride.java
package in.stackroute.umove.bookingservice.model;

import in.stackroute.umove.bookingservice.model.*;
import lombok.*;
import org.bson.types.ObjectId;
import org.springframework.data.mongodb.core.mapping.Document;

import javax.persistence.Column;
import javax.persistence.Id;
import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

/*
 * Vehicle booking details
 */

@Document(collection = "bookingList")
@Data
@NoArgsConstructor
@ToString
@Getter
@Setter
public class Ride
{
    @Id
    private UUID rideId;
    private RideStatus status;
    private User rider;
    private Zone sourceZone;
    private List<Zone> destinationZones;
    private Vehicle vehicle;
    private PaymentMethod paymentMethod;
    private LocalDateTime bookedAt;
    private LocalDateTime rideStartAt;
    private LocalDateTime rideEndAt;
    private Double distance;
    private int duration;
    private PaymentDetail paymentDetail;
    private PromoCode promoCode;

}
